tests/test_accelerated_op.py

```diff
@@ -0,0 +1,277 @@
+        assert torchopt.accelerated_op_available(torch.device('meta'))
+        assert torchopt.accelerated_op_available(['cpu', 'meta'])
+        assert torchopt.accelerated_op_available([torch.device('cpu'), torch.device('meta')])
+
+    if torch.cuda.is_available():
+        assert torchopt.accelerated_op_available()
+        assert torchopt.accelerated_op_available('cuda')
+        assert torchopt.accelerated_op_available('cuda:0')
+        assert torchopt.accelerated_op_available(0)
+        assert torchopt.accelerated_op_available(['cpu', 'cuda'])
+        assert torchopt.accelerated_op_available(['cpu', 'cuda:0'])
+        assert torchopt.accelerated_op_available(['cpu', 0])
+    else:
+        assert not torchopt.accelerated_op_available()
+        assert not torchopt.accelerated_op_available('cuda')
+        assert not torchopt.accelerated_op_available('cuda:0')
+        assert not torchopt.accelerated_op_available(0)
+        assert not torchopt.accelerated_op_available(['cpu', 'cuda'])
+        assert not torchopt.accelerated_op_available(['cpu', 'cuda:0'])
+        assert not torchopt.accelerated_op_available(['cpu', 0])
+
+
+@helpers.parametrize(
+    dtype=[torch.float64, torch.float32],
+    lr=[1e-2, 1e-3, 1e-4],
+    inplace=[True, False],
+)
+def test_accelerated_op(
+    dtype: torch.dtype,
+    lr: float,
+    inplace: bool,
+) -> None:
+    model, model_ref, model_base, loader = helpers.get_models(device='cpu', dtype=dtype)
+
+    fmodel, params, buffers = functorch.make_functional_with_buffers(model)
+    optim = torchopt.adam(
+        lr,
+        use_accelerated_op=True,
+    )
+    optim_state = optim.init(params)
+
+    fmodel_ref, params_ref, buffers_ref = functorch.make_functional_with_buffers(model_ref)
+    optim_ref = torchopt.adam(
+        lr,
+        use_accelerated_op=False,
+    )
+    optim_state_ref = optim_ref.init(params_ref)
+
+    for xs, ys in loader:
+        xs = xs.to(dtype=dtype)
+        pred = fmodel(params, buffers, xs)
+        pred_ref = fmodel_ref(params_ref, buffers_ref, xs)
+        loss = F.cross_entropy(pred, ys)
+        loss_ref = F.cross_entropy(pred_ref, ys)
+
+        grads = torch.autograd.grad(loss, params, allow_unused=True)
+        updates, optim_state = optim.update(grads, optim_state, params=params, inplace=inplace)
+        params = torchopt.apply_updates(params, updates, inplace=inplace)
+
+        grads = torch.autograd.grad(loss_ref, params_ref, allow_unused=True)
+        updates, optim_state_ref = optim.update(
+            grads, optim_state_ref, params=params_ref, inplace=inplace
+        )
+        params_ref = torchopt.apply_updates(params_ref, updates, inplace=inplace)
+
+    for p, p_ref in zip(params, params_ref):
+        helpers.assert_all_close(p_ref, p)
+
+
+def test_adam_bp_non_accelerated():
+    from torchopt.accelerated_op._src import adam_op
+
+    backup = torchopt.accelerated_op.adam_op.adam_op
+    torchopt.accelerated_op.adam_op.adam_op = adam_op
```

Comment:
Assign module attribute will not work as expected.

```suggestion
    torchopt.accelerated_op.adam_op.__dict__['adam_op'] = adam_op
```